shopfloor_single_product_transfer/services/single_product_transfer.py

```diff
@@ -550,31 +640,72 @@ def _find_user_move_line(self):
         domain = self._find_user_move_line_domain(user)
         return self.env["stock.move.line"].search(domain, limit=1)
 
-    def _set_quantity__by_product(self, move_line, barcode, confirmation=False):
-        product_handlers = [
-            self._set_quantity__scan_product,
-            self._set_quantity__scan_packaging,
-            self._set_quantity__scan_lot,
-        ]
-        product_response = self._use_handlers(product_handlers, move_line, barcode)
-        if product_response:
-            return product_response
-
-    def _set_quantity__by_location(self, move_line, barcode, confirmation=False):
-        search = self._actions_for("search")
-        location = search.location_from_scan(barcode)
-        handlers = [
-            # Cannot confirm if qty_done is not valid (> qty todo)
-            self._set_quantity__check_quantity_done,
+    def _set_quantity__by_location_handlers(self):
+        return [
             self._set_quantity__check_location,
             self._set_quantity__post_move,
         ]
-        if location:
+
+    def _set_quantity__by_location(self, move_line, location, confirmation=False):
+        # We're about to leave the `set_quantity` screen.
+        # First ensure that quantity is valid.
+        invalid_qty_response = self._set_quantity__check_quantity_done(move_line)
+        if invalid_qty_response:
+            return invalid_qty_response
+        move_line.result_package_id = False
+        handlers = self._set_quantity__by_location_handlers()
+        response = self._use_handlers(
+            handlers, move_line, location=location, confirmation=confirmation
```

Comment:
TypeError: _set_quantity__post_move() got multiple values for argument 'location'

```suggestion
            handlers, move_line, location, confirmation=confirmation
```

Reply:
done